cub/device/device_merge_sort.cuh

```diff
@@ -474,46 +679,90 @@ struct DeviceMergeSort
+   * cub::DeviceMergeSort::StableSortPairs(
+   *   d_temp_storage, temp_storage_bytes,
+   *   d_keys, d_values, num_items, custom_op);
    *
    * // Allocate temporary storage
    * cudaMalloc(&d_temp_storage, temp_storage_bytes);
    *
    * // Run sorting operation
-   * cub::DeviceMergeSort::StableSortPairs(d_temp_storage, temp_storage_bytes, d_keys, d_values, num_items, custom_op);
+   * cub::DeviceMergeSort::StableSortPairs(
+   *   d_temp_storage, temp_storage_bytes,
+   *   d_keys, d_values, num_items, custom_op);
    *
    * // d_keys      <-- [0, 3, 5, 6, 6, 8, 9]
    * // d_values    <-- [5, 4, 3, 1, 2, 0, 6]
-   *
-   * \endcode
-   *
-   * \tparam KeyIteratorT is a model of <a href="https://en.cppreference.com/w/cpp/iterator/random_access_iterator">Random Access Iterator</a>,
-   *         \p KeyIteratorT is mutable, and \p KeyIteratorT's \c value_type is
-   *         a model of <a href="https://en.cppreference.com/w/cpp/named_req/LessThanComparable">LessThan Comparable</a>,
-   *         and the ordering relation on \p KeyIteratorT's \c value_type is a <em>strict weak ordering</em>, as defined in the
-   *         <a href="https://en.cppreference.com/w/cpp/named_req/LessThanComparable">LessThan Comparable</a> requirements.
-   * \tparam ValueIteratorT is a model of <a href="https://en.cppreference.com/w/cpp/iterator/random_access_iterator">Random Access Iterator</a>,
-   *         \p ValueIteratorT is mutable, and \p ValueIteratorT's \c value_type is
-   *         a model of <a href="https://en.cppreference.com/w/cpp/named_req/LessThanComparable">LessThan Comparable</a>,
-   *         and the ordering relation on \p ValueIteratorT's \c value_type is a <em>strict weak ordering</em>, as defined in the
-   *         <a href="https://en.cppreference.com/w/cpp/named_req/LessThanComparable">LessThan Comparable</a> requirements.
-   * \tparam OffsetT is an integer type for global offsets.
-   * \tparam CompareOpT functor type having member <tt>bool operator()(KeyT lhs, KeyT rhs)</tt>
-   *         CompareOpT is a model of <a href="https://en.cppreference.com/w/cpp/concepts/strict_weak_order">Strict Weak Ordering</a>.
+   * @endcode
+   *
+   * @tparam KeyIteratorT
+   *   is a model of [Random Access Iterator], `KeyIteratorT` is mutable, and
+   *   `KeyIteratorT`'s `value_type` is a model of [LessThan Comparable],
+   *   and the ordering relation on `KeyIteratorT`'s `value_type` is a
+   *   *strict weak ordering*, as defined in the [LessThan Comparable]
+   *   requirements
+   *
+   * @tparam ValueIteratorT
+   *   is a model of [Random Access Iterator], `ValueIteratorT` is mutable, and
+   *   `ValueIteratorT`'s `value_type` is a model of [LessThan Comparable],
+   *   and the ordering relation on `ValueIteratorT`'s `value_type` is a
+   *   *strict weak ordering*, as defined in the [LessThan Comparable]
+   *   requirements
```

Comment:
Same here, values don't need to be comparable?

Reply:
Thank you for this note! I've fixed everything from #372. The only difference is that I've left the requirement for `ValueIteratorT` being modifiable.